tests/Services/Azure/Authentication/WindowHandleProvider.cs

```diff
@@ -60,9 +43,74 @@ public static IntPtr GetWindowHandle()
             }
             return IntPtr.Zero;
         }
-        else
+
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
```

Comment:
Thanks for the contribution, @danielmarbach 
The purpose of this window handle code is to ensure that the broker can appear above the z-order of the current application window. It is my understanding that MacOS didn't support the concept of inter-process window handles being used for z-order. Is there something I could be missing here?

As a side note, we don't actually support the broker on MacOS yet in any case. It should be coming soon though.

Reply:
I'm not super deep into this. When executing the tests on macos the interactive tests ran into the throwing of the platform exception part. After adding the platform specific part the interactive tests started working with the browser popping up and asking for the credentials. That's all I verified

Reply:
Thanks, makes sense. You are seeing the fallback to the interactive browser flow, not the broker. 

I think perhaps the best change to make short term would be to simply Assert.Ignore in the test if `true == RuntimeInformation.IsOSPlatform(OSPlatform.OSX)`

Reply:
That would mean updating all the usages of this class do have the Assert.Ignore, remove the macos branch here and leave the library import. Did I understand your proposal correctly?

Reply:
The only usage is the test, as far as I can see. I'm suggesting we revert the changes to WindowHandleProvider and assert.Ignore in the test that references this when running on MacOS. 

Once we support broker on MacOS, we'll need to figure out what the right thing to do here should be. As far as I understand, we would probably only supply IntPtr.Zero for MacOS, since the window handle won't be used by the broker APIs.

Reply:
Why should the source Gen library import usage be reverted? That seems a valuable change we would loose by reverting the class

Reply:
@christothes I pushed an update. Unfortunately, this project still targets xunit 2 and therefore doesn't allow adding skippable facts without taking a dependency to a community maintained project. Extending the existing fact attribute also felt a bit like scope creep...

Reply:
@christothes see https://github.com/Azure/azure-mcp/pull/31